pkgs/by-name/op/openvsp/package.nix

```diff
@@ -0,0 +1,114 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  git,
+  pkg-config,
+  python3,
+  eigen,
+  fltk,
+  glm,
+  glew,
+  cminpack,
+  libxml2,
+  graphviz,
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openvsp";
+  version = "3.40.1";
+
+  src = fetchFromGitHub {
+    owner = "OpenVSP";
+    repo = "OpenVSP";
+    rev = "OpenVSP_${version}";
+    hash = "sha256-PfXqnd75D06vER7X5w+VOm12mvLbcSe4sxmhtwBmPps=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    python3
+    git
+  ];
+
+  # swig & doxygen are not included as the build would fail since it tries to call
+  # "swig -doxygen" which fails Make as this is not a valid command.
+  # Seems like an upstream problem.
+  buildInputs = [
+    cminpack
+    eigen
+    fltk
+    glew
+    glm
+    graphviz
+    libxml2
+  ];
+
+  configurePhase = ''
+    mkdir -p build buildlibs
+
+    pushd buildlibs
+
+    cmake \
+      -DVSP_USE_SYSTEM_ADEPT2=false \
+      -DVSP_USE_SYSTEM_CLIPPER2=false \
+      -DVSP_USE_SYSTEM_CMINPACK=false \
+      -DVSP_USE_SYSTEM_CODEELI=false \
+      -DVSP_USE_SYSTEM_CPPTEST=false \
+      -DVSP_USE_SYSTEM_DELABELLA=false \
+      -DVSP_USE_SYSTEM_EIGEN=false \
+      -DVSP_USE_SYSTEM_EXPRPARSE=false \
+      -DVSP_USE_SYSTEM_FLTK=false \
+      -DVSP_USE_SYSTEM_GLEW=false \
+      -DVSP_USE_SYSTEM_GLM=false \
+      -DVSP_USE_SYSTEM_LIBIGES=false \
+      -DVSP_USE_SYSTEM_LIBXML2=false \
+      -DVSP_USE_SYSTEM_OPENABF=false \
+      -DVSP_USE_SYSTEM_PINOCCHIO=false \
+      -DVSP_USE_SYSTEM_STEPCODE=false \
+      -DVSP_USE_SYSTEM_TRIANGLE=false \
+      $src/Libraries \
+      -DCMAKE_BUILD_TYPE=Release
+
+    make -j$cores
+    popd
+
+    pushd build
+    cmake $src/src/ -DVSP_LIBRARY_PATH=$PWD/../buildlibs -DCMAKE_BUILD_TYPE=Release -DVSP_CPACK_GEN=DEB
```

Comment:
We could run this in preBuild

Reply:
Yes we could, but I fail to see how this isn't still part of configuring the environment.. Any reason?